### DISCUSSION TITLE
computed_field with create_model

### DISCUSSION THREAD

**Discussion by bitdivision** [pydantic/pydantic#7760] [Question]
I'm trying to work out how I can generate a pydantic schema using create_model while also using computed_fields.

i.e. given the following model:

```py
class MyModel(pydantic.BaseModel):
    thing: SomeEnum
    
    @pydantic.computed_field
    @property
    def derived_thing(self) -> bool:
        return self.thing == SomeEnum.abc
```

How would I create this using `create_model` without defining the computed fields manually? 

The use case is to setup various derived fields automatically for any enum fields defined on a model.

Thanks!

**Comment by c0x65o** [comment]
I came to ask the same question. I would like to know if it is possible.

**Comment by JMarkin** [comment]
Hello all,
It works for me
```python
from pydantic import create_model, computed_field, BaseModel
class Parent(BaseModel):
   pk: int

def id_(self) -> int:
   return self.pk

MyModel = create_model("MyModel", __base__=Parent, **{"id": computed_field(id_)})
MyModel(pk=22).model_dump()
# {'pk': 22, 'id': 22}
```

**Reply by villqrd** [reply]
This does not work anymore in 2.11.2 (works in 2.9.2)

AttributeError: 'property' object has no attribute '__mro__

**Comment by Viicos** [comment]
See https://github.com/pydantic/pydantic/issues/11709.
